network/book.go

```diff
@@ -123,13 +123,13 @@ func (b *Book) Sample(count uint, params ...interface{}) ([]string, error) {
 	defer b.mutex.Unlock()
 
 	// extract all the parameters for the sample
-	var filters []filterFunc
-	var sorts []sortFunc
+	var filters []func(e *entry) bool
+	var sorts []func(*entry, *entry) bool
```

Comment:
I don't think you need to change it here. In the type switch should be enough imo.